src/services/api/productService.ts

```diff
@@ -0,0 +1,22 @@
+import { BaseApiService } from "./base";
+import { AxiosResponse } from 'axios';
+import { IProduct } from '@/interfaces';
+export class ProductService {
+    
+    static getAllProducts = async (): Promise<AxiosResponse> => {
+        return await BaseApiService.instance.read("/products");
```

Comment:
Either embed this line in a try catch block or split this line into two. [ES-lint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) & [another example](https://jakearchibald.com/2017/await-vs-return-vs-return-await/#return-awaiting).

The same applies to all return await calls.